README.md

```diff
@@ -2,53 +2,179 @@
 
-Following instructions found at
+This add-on is meant to demonstrate integrations. Integrations can be written in
+any language or framework. We don't officially support this beyond that it's
+an example of how an add-on might be built.
+
+This add-on was built following instructions found at
 https://devcenter.heroku.com/articles/building-an-add-on.
 
-## Set up
+See [CONTRIBUTING.md](CONTRIBUTING.md) for information on how to set up and run
+the code for this application.
+
+## Add-on manifest
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#step-1-generate-your-add-on-manifest
+
+### Related code
+
+The manifest is not checked into version control because it contains secrets. A
+copy of the manifest with fake keys is located at
+[addon-manifest-example.json](addon-manifest-example.json). Although it is not
+checked into version control, it is helpful to know that the real manifest is
+named `addon-manifest.json` and will be called "the manifest" from this point
+forward.
+
+## Basic auth
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#basic-authentication
+
+### Related code
+
+Basic auth is implemented via the [`HttpBasicAuth`
+concern](app/controllers/concerns/http_basic_auth.rb). This concern is included
+in [`ApplicationController`](app/controllers/application_controller.rb), which
+all other controllers inherit from, because the Add-on Partner API sends basic
+auth credentials with all requests. Basic auth is selectively skipped for the
+single sign-on views, which are accessed by Heroku customers who are using the
+add-on rather than the Add-on Partner API.
+
+The username and password for basic auth are being accessed via environment
+variables whose values match the `slug` and `password` fields in the manifest.
+
+## The provisioning request
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#the-provisioning-request
+* https://devcenter.heroku.com/articles/getting-started-with-asynchronous-provisioning#implementation
+* https://devcenter.heroku.com/articles/add-on-partner-api-reference#add-on-action-create-provision
+* https://devcenter.heroku.com/articles/platform-api-reference#add-on-action-provision
+
+### Related code
+
+The Sudo Sandwich add-on is provisioned synchronously or asynchronously
+depending on the plan that is selected. This somewhat artificial plan slug
+distinction is just to demonstrate the process of provisioning plans
+asynchronously and synchronously.
+
+For both types of provisioning, the `base_url` key in the add-on manifest
+specifies a path of `/heroku/resources`, which is routed to the
+[`Heroku::ResourcesController`](app/controllers/heroku/resources_controller.rb)
+`#create` method in this codebase. This is the endpoint that Heroku hits when a
+customer creates an add-on resource.
+
+#### Synchronous provisioning
+
+The controller looks for the `plan` param, and if it matches
```

Comment:
Even synchronous provisioning should probably exchange the oauth grant for an access token. That way if they need to update config for example rotating credentials in the future they will be able to use the Platform API for Partners to do so.

Reply:
I am retrieving an access_token for both type of provisioning but will make that clearer here

Reply:
updated at 924eac6